mhkit/qc/upcrossing.py

```diff
@@ -1,3 +1,52 @@
+"""
+Upcrossing Analysis Functions
+=============================
+This module contains a collection of functions that facilitate upcrossing 
+analyes.
+
+Key Functions:
+--------------
+- `upcrossing`: Finds the zero upcrossing points.
+  
+- `peaks`: Finds the peaks between zero crossings.
+  
+- `troughs`: Finds the troughs between zero crossings.
+  
+- `heights`: Calculates the height between zero crossings.
+  
+- `periods`: Calculates the period between zero crossings.
+  
+- `custom`: Fetches specified wind data parameters for given 
+  latitude-longitude points and years from the WIND Toolkit hindcast dataset. 
+  Supports caching for faster repeated data retrieval.
+  
+Dependencies:
+-------------
+- rex: Library to handle renewable energy datasets.
+- pandas: Data manipulation and analysis.
+- os, hashlib, pickle: Used for caching functionality.
+- matplotlib: Used for plotting.
+
+Notes:
+------
+- To access the WIND Toolkit hindcast data, users need to configure `h5pyd` 
+  for data access on HSDS (see the WTK_hindcast_example notebook for more details).
+
+- While some functions perform basic checks (e.g., verifying that latitude 
+  and longitude are within a predefined region), it's essential to understand 
+  the boundaries of each region and the available parameters and elevations in the dataset.
```

Comment:
Some of these comments don't appear relevant to this module (eg references to configuring h5pyd), just checking that we want to keep them here?